app/controllers/posts_controller.rb

```diff
@@ -24,7 +33,7 @@ def edit
   # POST /posts
   # POST /posts.json
   def create
-    @post = Post.new(post_params)
+    @post = Post.new(post_params.merge(user_id: current_user.id))
```

Comment:
if your relations are correct, you have to change your names `user_id` for `user` and `current_user.id` for `current_user`, this naming shouln't exist in any part of your code.